### PR TITLE
Refactor Memo List Serialization

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Refactor memo list to be standardized. 

### Context of Change

Refactor method to use same idioms as other list serialization methods (see for example, `pathListToJSON`). 

Specifically:
- Change name to `memoListToJSON` for consistency
- Use a simple map function internally
- Do not check inputs for validity, defer that responsibility to callers (note, this removes a unit test)
- Input is `Array<Memo>` and output is `Array<MemoJSON`>

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

No change. This is prep work for a larger refactor of common transaction fields in #519 

## Test Plan

CI shows that this does not affect the public API; removed a unit test which was testing removed functionality.

### CODE REVIEW

File: src/XRP/serializer.ts [hunk 1]
```diff
@@ -805,13 +808,9 @@ const serializer = {
...
-  memosToJSON(memos: Memo[]): { Memos: MemoJSON[] } | undefined {
-    if (!memos.length) {
-      return undefined
-    }
-
-    const convertedMemos = memos.map((memo) => this.memoToJSON(memo))
-    return { Memos: convertedMemos }
+  memoListToJSON(memos: Memo[]): MemoJSON[] {
+    // eslint-disable-next-line @typescript-eslint/unbound-method -- Manually assigning `this`.
+    return memos.map(this.memoToJSON, this)
```
Comment (tedkalaw) [inline src/XRP/serializer.ts:813]: can you use an arrow function instead here?

```
return memos.map((memo) => this.memoToJSON(memo))
```

`this` is presered in arrow functions

Comment (keefertaylor) [reply src/XRP/serializer.ts:813]: Totally. I was trying to make the linter happy here (we do this other places too). 

Do we have any thoughts on whether we should disable this linter rule?

**Comment by keefertaylor** [comment]
/+ Hans who may have comments about the linter. I think you've mentioned we could disable the rule in question in the past, I just forget exactly what you've said :)

**Comment by keefertaylor** [comment]
Gentle ping, @tedkalaw

**Review by amiecorso — APPROVED** [review]

File: src/XRP/serializer.ts (hunk 1 above)
Comment (tedkalaw) [reply src/XRP/serializer.ts:813]: it gets mad with arrow functions? the dox for the rule implies that arrow funcs are cool: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/unbound-method.md

if we're doing this other places already, i say just go for it as is

**Review by tedkalaw — APPROVED** [review]

**Pull request merged**